### PR TITLE
feat(ui-tests): scaffolds basic test

### PR DESCRIPTION
**Pull request by arein**
Basic tests covering subscribe/unsubscribe.

# Description

Please include a brief summary of the change.

# Type of change

Please put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Fixes/Resolves (Optional)

Please list any issues that are fixed by this PR in the format `#<issue number>`. If it fixes multiple issues, please put each issue in a separate line. If this Pull Request does not fix any issues, please delete this section.

# Examples/Screenshots (Optional)

Please attach a screenshot or screen recording of features/fixes you've implemented to verify your changes. Please also note any relevant details for your configuration. If your changes do not affect the UI, please delete this section.

Use this table template to show examples of your changes:

| Before       | After        |
| ------------ | ------------ |
| Content Cell | Content Cell |
| Content Cell | Content Cell |

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +58,7 @@
     "@vitejs/plugin-react": "^2.2.0",
     "@walletconnect/types": "2.11.0",
     "autoprefixer": "^10.4.14",
+    "dotenv": "16.3.1",
```
Comment (chris13524) [inline package.json:61]: Why was this added? I see the `require('dotenv').config();` line was removed below

```suggestion
```

**Review by chris13524 — CHANGES_REQUESTED** [review]
`yarn.lock` shouldn't need to change, can you rebase or whatever?

Also Vercel lint is failing.

**Comment by arein** [comment]
> `yarn.lock` shouldn't need to change, can you rebase or whatever?
> 
> Also Vercel lint is failing.

Vercel is failing due to

``

src/components/settings/PrivacySettings/index.tsx(44,10): error TS2741: Property 'children' is missing in type '{ title: string; subtitle: string; }' but required in type 'IProps'.
--
08:56:06.184 | error Command failed with exit code 2.
```

Didn't touch this

**Comment by chris13524** [comment]
@devceline can you fix this PR?

File: src/Modals.tsx [hunk 2]
```diff
@@ -96,7 +97,7 @@ export const Modals = () => {
 
       {shouldShowPWAModal && <PwaModal />}
 
-      {isNotificationPwaModalOpen && <NotificationPwaModal />}
+      {!isCI && isNotificationPwaModalOpen && <NotificationPwaModal />}
```
Comment (chris13524) [inline src/Modals.tsx:100]: I think we can avoid this flag by [granting the `notification` permission](https://playwright.dev/docs/api/class-browsercontext#browser-context-grant-permissions). Also we'll need to test these states anyway so this will have to change.

File: tests/shared/pages/InboxPage.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { type Locator, type Page, expect } from '@playwright/test'
+
+import { BASE_URL } from '../constants'
+
+export class ModalPage {
```
Comment (chris13524) [inline tests/shared/pages/InboxPage.ts:5]: Rename to `InboxPage`?

File: tests/shared/pages/InboxPage.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    if (!isVisible) return
+    await this.page.locator('.NotificationPwaModal__close-button').first().click()
+  }
+
+  async subscribe(nth: number) {
+    await this.page.locator('.AppCard__body > .AppCard__body__subscribe').nth(nth).click()
+    await this.page.getByText('Subscribed to', { exact: false }).isVisible()
+  }
+
+  async unsubscribe(nth: number) {
```
Comment (chris13524) [inline tests/shared/pages/InboxPage.ts:48]: This API will flake in future revisions because number is non-deterministic. Better to select by name.

**Review by chris13524 — APPROVED** [review]
LGTM other than some suggestions for future iterations
